### PR TITLE
Bugs/48300 Bundling and minification multiinstance issues

### PR DESCRIPTION
**Pull request by vadym-hyryn**

### CODE REVIEW

File: Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/DocumentProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+                    {
+                        Log.LogWarning(AppNameForLogs, $"Could not compile styles for the source {source}");
+                        return;
+                    }
+                    bundleObj.Include(compiledStylePath);
+                }
+
+                if (notSupportedExtentions.Any())
+                {
+                    string message = $"Styles with the following extensions have not available CSS decompilers: " +
```
Comment (napernik) [inline Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/DocumentProcessor.cs:230]: *compilers

File: Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/BundleMinifyHttpModule.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+using System.Web.Optimization;
+using Composite.Core;
+using static Orckestra.Web.BundlingAndMinification.CommonValues;
+
+namespace Orckestra.Web.BundlingAndMinification
+{
+    public class BundleMinifyHttpModule : IHttpModule
+    {
+        private static ConcurrentBag<string> _bundlesRequests = new ConcurrentBag<string>();
+        private const string _bundlePathRegex = "^/Bundles/(Scripts|Styles)_([a-zA-Z0-9]{32})?v=(.+?)$";
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/BundleMinifyHttpModule.cs]: Thanks, updated to compiled regex

File: Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification.Package/Package/Web.config/Install.xsl [hunk 3]
```diff
@@ -21,6 +21,11 @@
           </dependentAssembly>
         </assemblyBinding>
       </runtime>
+      <system.webServer>
+        <modules runAllManagedModulesForAllRequests="true">
+          <add name="BundleMinifyHttpModule" type="Orckestra.Web.BundlingAndMinification, Orckestra.Web.BundlingAndMinification" />
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification.Package/Package/Web.config/Install.xsl]: Fixed type

File: Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/BundleMinifyHttpModule.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        {
+            application.BeginRequest += (a, b) => context_BeginRequest(application.Context);
+        }
+
+        private void context_BeginRequest(HttpContext context)
+        {
+            if (!BundleMinifyStyles && !BundleMinifyStyles) return;
+            var requestPath = context.Request.Path;
+
+            var match = Regex.Match(requestPath, _bundlePathRegex);
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/BundleMinifyHttpModule.cs]: Thanks, updated regex

File: Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/BundleMinifyHttpModule.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+            Bundle bundleObj = BundleTable.Bundles.GetBundleFor(requestPath);
+            if (bundleObj !=null)
+            {
+                _bundlesRequests.Add(hash);
+                return;
+            }
+
+            var physicalPath = HostingEnvironment.MapPath(BundlesCacheFolder);
+            ProcessCache(actionType, physicalPath);
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/BundleMinifyHttpModule.cs]: Agree, thank you for noticing

File: Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs [hunk 6]
```diff
@@ -28,17 +20,20 @@ public void Filter(XhtmlDocument document, IPage page)
...
 
-            if ((!BundleMinifyScripts && !BundleMinifyStyles)
-                || IsAdminConsoleRequest(httpContext)
-                || (httpContext.Request["c1mode"] != "perf" &&
-                (httpContext.IsDebuggingEnabled       
-                || UserValidationFacade.IsLoggedIn())))
-            {
-                return;
-            }
+            //if (
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs]: unintentionally, commented this section for testings, thanks

File: Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/DocumentProcessor.cs (hunk 1 above)
Comment (vadym-hyryn) [reply Orckestra.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/DocumentProcessor.cs]: Updated

**Review by napernik — APPROVED** [review]

**Pull request merged**